src/components/Reports/SalaryCalculator/MhaRequestSection/useMhaRequestData.ts

```diff
@@ -0,0 +1,120 @@
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+import { useLocale } from 'src/hooks/useLocale';
+import { currencyFormat } from 'src/lib/intlFormat';
+import { amount } from 'src/lib/yupHelpers';
+import { useSalaryCalculator } from '../SalaryCalculatorContext/SalaryCalculatorContext';
+
+export const useMhaRequestData = () => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { calculation, hcm } = useSalaryCalculator();
+
+  const self = useMemo(() => hcm?.[0], [hcm]);
+  const spouse = useMemo(() => hcm?.[1], [hcm]);
```

Comment:
Property accesses are cheap, so I don't think you need to memoize this.

```suggestion
  const self = hcm?.[0];
  const spouse = hcm?.[1];
```